lib/sdam/topology.js

```diff
@@ -101,16 +112,34 @@ class Topology extends EventEmitter {
    * @param {number} [options.localThresholdMS=15] The size of the latency window for selecting among multiple suitable servers
    * @param {number} [options.serverSelectionTimeoutMS=30000] How long to block for server selection before throwing an error
    * @param {number} [options.heartbeatFrequencyMS=10000] The frequency with which topology updates are scheduled
+   * @param {MongoCredentials} [options.credentials] Authentication credentials.
+   * @param {any} [options.cursorFactory] a Cursor constructor class.
```

Comment:
Why are we using `any` for all of these types? Was it just exhaustion at dealing with the codebase? I think a number of them are pretty explicitly some other type (`cursorFactory` is probably a `CoreCursor`, etc).

Reply:
Because of that exports issue, `CoreCursor` is not constructable. Yes exhaustion factored in too 🙏 but this unfortunately is just a type error in the current state
I could give it a better type annotation like here: (maybe hoist it to a typedef as per your comment for this line)
https://github.com/mongodb/node-mongodb-native/blob/a244966dc624371bed4a4b13530b30ad843bab16/lib/sdam/topology.js#L737

Reply:
But `CoreCursor` isn't the only one here, `maxPoolSize` and `minHeartbeatFrequencyMS` are both `number`, `reconnect` is a `boolean`, `logger` is a `Logger` instance, etc.